ormar/models/model.py

```diff
@@ -242,7 +243,19 @@ async def update(self: T, _columns: List[str] = None, **kwargs: Any) -> T:
         self_fields = self._extract_model_db_fields()
         self_fields.pop(self.get_column_name_from_alias(self.Meta.pkname))
         if _columns:
-            self_fields = {k: v for k, v in self_fields.items() if k in _columns}
+            onupdate_fields = self.get_fields_with_onupdate()
+            self_fields = {
+                k: v
+                for k, v in self_fields.items()
+                if k in _columns or k in onupdate_fields
+            }
+        if not kwargs and not _columns:
```

Comment:
If all are null here, the update is based on user-modified attributes on the instance, with user-modified attributes taking precedence over on_update.